engine/src/main/java/org/hibernate/validator/path/IndexedPath.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.validator.path;
+
+import org.hibernate.validator.Incubating;
+
+/**
+ * An extended representation of the validation path, provides Hibernate Validator specific functionality.
+ * Represents a path with access to the nodes by their index.
+ *
+ * @since 9.1
+ */
+@Incubating
+public interface IndexedPath extends Path {
```

Comment:
@yrodiere do you think `IndexedPath` would be a good name for a path that allows access to the nodes by an index? (or could someone think that this path is to be stored in a Lucene index 😁 )

Reply:
I'm not sure why you didn't just pick `HibernatePath`, similar to what is being done for every other Jakarta Validation interface that we extend in Hibernate Validator?

If this really needs to be about the access by index, maybe `RandomAccessPath`, similar to `RandomAccessList`? Or maybe try to find some inspiration at https://openjdk.org/jeps/431 ?

Reply:
> I'm not sure why you didn't just pick HibernatePath

right ... well we sort of have two in HV -- one using a link approach and the other one backed by an array ... 
so `ArrayPath` `LinkedPath` as with lists ? 🙈

Reply:
Do you need two?

Serious question, as performance wise I've been talked `ArrayList` often outperforms `LinkedList` anyway, so...